asllib/StaticEnv.ml

```diff
@@ -218,6 +225,15 @@ let add_local x ty ldk env =
       };
   }
 
+let add_local_immutable_expr x e env =
```

Comment:
Perhaps add a comment that it is **assume** that `e` already refers to the oldest expressions?
Actually, maybe it's better to take `e` and substitute existing expressions for its variables here instead of in `annotate_local_decl_item`?